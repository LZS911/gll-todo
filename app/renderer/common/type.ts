import { SizeEnum } from './enum';

export type Dictionary = {
  [key: string]: string | number | boolean | Dictionary | string[] | undefined;
};
export interface IFormRules {
  required?: boolean;
  message?: string;
  validate?: (rules: IFormRules[], value: string) => Promise<string>;
  trigger?: string | string[];
}

export interface IComponentCommonProps {
  className?: string;
  size?: keyof typeof SizeEnum;
  name?: string;
}

export interface IWarnProps {
  title?: string;
  content?: string;
  onCancel?: () => void;
  cancelText?: string;
  onOk: () => void;
  okText?: string;
  okButtonProps?: string;
  closable?: boolean;
}
