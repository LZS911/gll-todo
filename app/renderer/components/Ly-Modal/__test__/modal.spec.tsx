import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import * as React from 'react';
import LyModal from '..';
import { ILyModalProps } from '../index.type';

describe('test ly modal', () => {
  afterEach(() => {
    cleanup();
  });
  it('should render modal title', () => {
    const props: ILyModalProps = {
      title: 'test modal',
      closeModal: () => {},
      onSave: () => {},
    };
    const { container } = render(<LyModal {...props} />);

    screen.getByText(props.title);
    expect(container).toMatchSnapshot();
  });

  it('should render exit by default model', async () => {
    const props: ILyModalProps = {
      title: 'test modal',
      closeModal: () => {},
      onSave: () => {},
    };
    const { container, rerender } = render(<LyModal {...props} />);
    screen.getByTestId('modal-exit-icon');
    screen.getByTestId('ly-button-modal-cancel-btn');

    rerender(<LyModal {...props} showExit={true} />);
    screen.getByTestId('modal-exit-icon');
    screen.getByTestId('ly-button-modal-cancel-btn');

    rerender(<LyModal {...props} showExit={false} />);
    expect(container).toMatchSnapshot();
  });

  it('should correct execution confirm and cancel the callback', () => {
    const closeModal = jest.fn();
    const onSave = jest.fn();
    const props: ILyModalProps = {
      title: 'test modal',
      closeModal,
      onSave,
    };
    render(<LyModal {...props} />);
    expect(closeModal).toBeCalledTimes(0);
    expect(onSave).toBeCalledTimes(0);

    fireEvent.click(screen.getByTestId('ly-button-modal-confirm-btn'));
    expect(onSave).toBeCalledTimes(1);

    fireEvent.click(screen.getByTestId('ly-button-modal-cancel-btn'));
    expect(closeModal).toBeCalledTimes(1);
  });

  it('should add className', () => {
    const className = 'test-modal-class';
    const props: ILyModalProps = {
      title: 'test modal',
      closeModal: () => {},
      onSave: () => {},
      className,
    };
    const { container } = render(<LyModal {...props} />);
    expect(container).toMatchSnapshot();
    expect(
      (
        container.firstElementChild?.lastChild?.firstChild as HTMLElement
      ).classList.contains(className)
    ).toBeTruthy();
  });
});
