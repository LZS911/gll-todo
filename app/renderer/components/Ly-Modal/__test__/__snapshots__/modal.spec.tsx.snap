// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ly modal should add className 1`] = `
<div>
  <div
    class="ly-modal-root"
    data-testid="ly-modal"
  >
    <div
      class="ly-modal-mask"
    />
    <div
      class="ly-modal-wrap animate__animated animate__zoomIn"
    >
      <div
        class="ly-modal test-modal-class"
      >
        <div
          class="ly-modal-content"
        >
          <div
            class="ly-modal-content-title"
          >
            test modal
          </div>
          <i
            class="ly-modal-content-icon"
            data-testid="modal-exit-icon"
          >
            x
          </i>
          <div
            class="ly-modal-content-children"
          />
          <div
            class="ly-modal-content-footer"
          >
            <button
              class="ly-button "
              data-testid="ly-button-modal-cancel-btn"
            >
              取消
            </button>
            <button
              class="ly-button "
              data-testid="ly-button-modal-confirm-btn"
            >
              确认
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test ly modal should render exit by default model 1`] = `
<div>
  <div
    class="ly-modal-root"
    data-testid="ly-modal"
  >
    <div
      class="ly-modal-mask"
    />
    <div
      class="ly-modal-wrap animate__animated animate__zoomIn"
    >
      <div
        class="ly-modal"
      >
        <div
          class="ly-modal-content"
        >
          <div
            class="ly-modal-content-title"
          >
            test modal
          </div>
          <div
            class="ly-modal-content-children"
          />
          <div
            class="ly-modal-content-footer"
          >
            <button
              class="ly-button "
              data-testid="ly-button-modal-confirm-btn"
            >
              确认
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test ly modal should render modal title 1`] = `
<div>
  <div
    class="ly-modal-root"
    data-testid="ly-modal"
  >
    <div
      class="ly-modal-mask"
    />
    <div
      class="ly-modal-wrap animate__animated animate__zoomIn"
    >
      <div
        class="ly-modal"
      >
        <div
          class="ly-modal-content"
        >
          <div
            class="ly-modal-content-title"
          >
            test modal
          </div>
          <i
            class="ly-modal-content-icon"
            data-testid="modal-exit-icon"
          >
            x
          </i>
          <div
            class="ly-modal-content-children"
          />
          <div
            class="ly-modal-content-footer"
          >
            <button
              class="ly-button "
              data-testid="ly-button-modal-cancel-btn"
            >
              取消
            </button>
            <button
              class="ly-button "
              data-testid="ly-button-modal-confirm-btn"
            >
              确认
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
